tests/cases/fourslash/codeFixPromiseToAsyncCatchAndRej.ts

```diff
@@ -0,0 +1,34 @@
+/// <reference path='fourslash.ts' />
+
+// @target: es6
+
+////function [|f|]():Promise<void> {
+////    return fetch('http://yahoo.com').then(result => { console.log(result); }, rejection => { console.log("rejected:", rejection); }).catch(err => { console.log(err) });
+////}
+
+verify.getSuggestionDiagnostics([{
+    message: "This may be converted to use async and await.",
+    code: 80006,
+}]);
+
+
+verify.codeFix({
+    description: "Convert to use async and await",
+    index: 0,
+    newFileContent:
+`async function f():Promise<void> {
+    try {
+        let result;
+        try {
+            result = await fetch('http://yahoo.com');
+        }
+        catch (rejection) {
+            console.log("rejected:", rejection);
+        }
+        console.log(result);

```

Comment:
This is incorrect because you would end up calling `console.log(result)` even if `fetch` fails.

The only way to truly model the same semantics would be something like:

```ts
try {
    let result;
    label: {
        try {
            result = await fetch('http://yahoo.com');
        }
        catch (rejection) {
            console.log("rejected:", rejection);
            break label;
        }
        console.log(result);
    }
}
catch (err) {
    console.log(err);
}
```